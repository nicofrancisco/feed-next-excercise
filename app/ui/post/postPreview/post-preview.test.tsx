import '@testing-library/jest-dom';
import React from "react";
import { render, screen } from "@testing-library/react";
import PostPreview from "./post-preview";
import { Post, User } from "../../../lib/types";
import { expect } from 'vitest';


const mockPost: Post = {
  id: 1,
  title: "Test Post",
  body: "This is a Post Preview test",
  userId: 0,
};

const mockAuthor: User = {
  id: 1,
  name: "John Doe",
  username: "",
  email: "",
  address: {
    street: "",
    suite: "",
    city: "",
    zipcode: "",
    geo: {
      lat: "",
      lng: "",
    },
  },
  phone: "",
  website: "",
  company: {
    name: "",
    catchPhrase: "",
    bs: "",
  },
};

describe("PostPreview Component", () => {
  it("should render with author information", () => {
    render(
      <PostPreview post={mockPost} user={mockAuthor} />
    );
    expect(screen.getByText(mockAuthor.name)).toBeTruthy();
    expect(screen.getByText(mockPost.title)).toBeTruthy();
    expect(screen.queryByText(new RegExp(mockPost.body))).toBeTruthy();
  });

  it("should render without author information", () => {
    render(<PostPreview post={mockPost} user={null} />);
    expect(
      screen.getByText(/Sorry, no author information available/i)
    ).toBeTruthy();
  });

  it("should render correctly with given props and create snapshot", () => {
    const { asFragment } = render(
      <PostPreview post={mockPost} user={mockAuthor} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
