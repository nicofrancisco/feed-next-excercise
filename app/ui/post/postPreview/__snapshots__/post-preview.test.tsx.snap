// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostPreview Component > should render correctly with given props and create snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="post-preview"
  >
    <a
      aria-label="Read post: "Test Post" by John Doe"
      class=" px-5 py-2 flex justify-between hover:bg-gray-100 sm:px-7"
      href="/post?id=1"
    >
      <article
        class=" flex space-x-4"
      >
        <div
          class=" flex space-x-4"
        >
          <div
            class="min-w-min flex flex-col items-center row-start-1 row-end-7 col-start-1 col-end-3"
          >
            <img
              alt="user-avatar"
              class="min-h-14 min-w-14 flex-none rounded-full bg-gray-50"
              loading="lazy"
              src="https://api.dicebear.com/8.x/adventurer/svg?seed=John Doe"
            />
            <div
              class="grow border-l-4 border-slate-200 min-h-5 w-0 mt-2"
            />
          </div>
          <div
            class="flex flex-col"
          >
            <div
              class="h-16 flex flex-col justify-center"
            >
              <span
                class="font-bold text-lg"
              >
                John Doe
              </span>
              <time
                class="text-xs leading-5 text-gray-500"
              >
                Old record
              </time>
            </div>
            <h2
              class="text-md font-semibold leading-6 text-gray-700 capitalize"
            >
              Test Post
            </h2>
            <p
              class="mt-2 line-clamp-3 max-h-32 text-wrap overflow-y-hidden text-ellipsis overflow-hidden text-sm leading-6 text-gray-800"
            >
              This is a Post Preview test
            </p>
          </div>
        </div>
      </article>
    </a>
  </div>
</DocumentFragment>
`;
